src/main/java/com/mongodb/hibernate/jdbc/MongoStatement.java

```diff
@@ -228,21 +279,379 @@ void checkClosed() throws SQLException {
+
+        private static WriteModel<BsonDocument> createInsertModel(BsonDocument insertDocument) {
+            return new InsertOneModel<>(insertDocument);
+        }
+
+        private static WriteModel<BsonDocument> createUpdateModel(
+                BsonDocument updateStatement, CommandDescription commandDescription)
+                throws SQLFeatureNotSupportedException {
+            checkStatementFields(updateStatement, commandDescription, SUPPORTED_UPDATE_STATEMENT_FIELDS);
+            var isMulti = updateStatement.getBoolean("multi", FALSE).getValue();
+            var filter = updateStatement.getDocument("q");
+            var updateModification = updateStatement.get("u");
+            if (updateModification == null) {
+                // We force exception here because the field is mandatory.
+                updateStatement.getDocument("u");
+            }
+            if (!(updateModification instanceof BsonDocument updateDocument)) {
+                throw new SQLFeatureNotSupportedException("Only document type is supported as value for field: [u]");
+            }
+            if (isMulti) {
+                return new UpdateManyModel<>(filter, updateDocument);
+            }
+            return new UpdateOneModel<>(filter, updateDocument);
+        }
+
+        private static WriteModel<BsonDocument> createDeleteModel(
+                BsonDocument deleteStatement, CommandDescription commandDescription)
+                throws SQLFeatureNotSupportedException {
+            checkStatementFields(deleteStatement, commandDescription, SUPPORTED_DELETE_STATEMENT_FIELDS);
+            var isSingleDelete = deleteStatement.getNumber("limit").intValue() == 1;
+            var filter = deleteStatement.getDocument("q");
+
+            if (isSingleDelete) {
+                return new DeleteOneModel<>(filter);
+            }
+            return new DeleteManyModel<>(filter);
+        }
+
+        private static void checkStatementFields(
+                BsonDocument statement, CommandDescription commandDescription, Set<String> supportedStatementFields)
+                throws SQLFeatureNotSupportedException {
+            checkFields(
+                    commandDescription,
+                    UNSUPPORTED_MESSAGE_STATEMENT_FIELD,
+                    supportedStatementFields,
+                    statement.keySet().iterator());
+        }
+
+        private static void checkCommandFields(
+                BsonDocument command, CommandDescription commandDescription, Set<String> supportedCommandFields)
+                throws SQLFeatureNotSupportedException {
+            var iterator = command.keySet().iterator();
+            iterator.next(); // skip the command name
+            checkFields(commandDescription, UNSUPPORTED_MESSAGE_COMMAND_FIELD, supportedCommandFields, iterator);
+        }
+
+        private static void checkFields(
+                CommandDescription commandDescription,
+                String exceptionMessage,
```

Comment:
This is a template, not a message: `exceptionMessage` -> `exceptionMessageTemplate`.